wikiscraper.py

```diff
@@ -1,175 +1,264 @@
-                if var != "":
-                    data[ext["target"]["name"]] = var
-        elif sel["type"] == "collectedTemplateVariables":
-            for i in xrange(sel["rangeStart"], sel["rangeEnd"]+1):
-                varName = "{0}{1}".format(sel["value"], i)
-                targetName = ext["target"]["name"]
-                if template.has(varName):
-                    if targetName not in data:
-                        data[targetName] = []
-                    data[targetName].append(template.get(varName).value.strip())
-
-    # TODO: Make this configurable too
-    hashPageText(data)
-    combineNamesAndDescriptions(data, "path")
-    combineNamesAndDescriptions(data, "queryString")
-    combineNamesAndDescriptions(data, "jsonBody")
-    cleanExampleRequest(data)
-    cleanExampleResponse(data)
-
-    return data
-
-# TODO: Fucking awful
-def handleResponse(response, baseUrl, extractedData=[]):
-    for pageId, pageData in rget(response, ["query", "pages"]).iteritems():
-        extractedData.append(performExtractions(pageData))
-
-    if "query-continue" in response:
-        continueParam = response["query-continue"].values()[0]
-        url = baseUrl + "&%s=%s" % (continueParam.keys()[0], continueParam.values()[0])
-        nextResponse = json.loads(urllib2.urlopen(url).read())
-        extractedData = handleResponse(nextResponse, baseUrl, extractedData)
-
-    return extractedData
-
-
-# TODO: Check out argparse
-# Make sure a config file's name was passed in, then load it
-if len(sys.argv) == 2:
-    # TODO: Verify file exists
-    conf = yaml.load(file(sys.argv[1], "r"))
+            for phrase in phrases:
+                res = res.replace(phrase, "")
+            try:
+                res = json.loads(res)
+                data["exampleResponses"] = [json.dumps(res)]
+                del data["exampleResponse"]
+            except ValueError as ve:
+                print "Error decoding JSON for exampleResponse in " + data["name"]
+                print res
+
+    # TODO: This is VERY specific
+    def performExtractions(self, pageData):
+        text = rget(pageData, ["revisions", 0, "*"])
+        # TODO: Make this configurable
+        text = self.stripSyntaxHighlighter(text)
+        wikicode = mwparserfromhell.parse(text)
+        template = wikicode.filter_templates()[0]
+        # Perform extractions
+        data = {}
+        for ext in self.conf["extractions"]:
+            sel = ext["selector"]
+            if sel["type"] == "pageData":
+                data[rget(ext, ["target", "name"])] = pageData[sel["value"]]
+            elif sel["type"] == "pageText":
+                data[rget(ext, ["target", "name"])] = rget(pageData, ["revisions", 0, "*"])
+            elif sel["type"] == "templateVariable":
+                if template.has(sel["value"]):
+                    var = template.get(sel["value"]).value.strip()
+                    if var != "":
```

Comment:
If you can guarantee var is of type `String` (strings are falsey in Python) you could also say:
```python
if var:
    data[ext["target"]["name"]] = var
```